### DISCUSSION TITLE
Subject: Connection Issue

### DISCUSSION THREAD

**Discussion by AHirtCodes** [supabase/supabase#29579] [Questions]
Dear Supabase Support Team,

I am writing to seek assistance with a connection issue I'm experiencing with my Supabase database. Despite successful network connectivity tests, I'm unable to connect to the database through my application or Prisma. Here are the details:

Issue Description:

I can successfully establish a TCP connection to the Supabase database server on port 5432 using the netcat (nc) command.
However, I'm encountering issues when trying to connect through my application or Prisma.
Steps Taken:

Verified network connectivity using netcat command.
Checked authentication details and connection string format.
Reviewed SSL/TLS settings.
Double-checked Prisma configuration.
Verified database permissions.
Questions:

Given the successful netcat test, what could be preventing my application or Prisma from connecting?
Are there any known issues or common misconfigurations that might cause this behavior?
What additional troubleshooting steps would you recommend?
I appreciate any guidance you can provide to resolve this issue. Please let me know if you need any additional information.

Thank you for your time and assistance.

**Comment by GaryAustin1** [comment]
Supabase Support is only available by contacting thru the support tab in your instance or email.

Are you using the Supavisor ipv4 compatible connection or the direct database port connection?
Prisma requires two connections, one with the 6543 port for "user" operations and 5342 for migrations and the like.  There are settings required for the 6543 port.  https://supabase.com/partners/integrations/prisma is the guide for setting those.

Also for Prisma as you are passing a string, if the password has special characters the entire string will not be correct without escaping them.

**Comment by AHirtCodes** [comment]
I am using the Supavisor ipv4 connection, does that make a difference? I
have set up the 2 different connections (6543) for "user" and (5342) for
"migrations"
